Mds/new mobile sdk/okta 503880 revise quickstart (#3424)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md

> To check for an existing session may take two steps:

Reviewer: If these steps are required, are they both required? If not, it's probably better to use bullet points.

Author: Nice catch! The numbered steps were from the older version which required both steps. I've updated them to bullets. That will appear on the next checkin.

> To check for an existing session may take two steps:

Reviewer 2: ```suggestion
When you check for an existing session, it may take two steps:
```

Author: Thanks. In the next commit.

> 1. Check for an existing valid token.

Author: Change made based on earlier comment, though I used "-" instead of "*" for the bullet... paranoia since "*" is also used for emphasis and bold.

> 1. Check for an existing valid token.

Reviewer: True :)

> 1. Refresh the token if it has expired.

Author: Same comment as above.

> - Refresh the token if it has expired.

Reviewer 2: ```suggestion
- Refresh the token if it's expired.
```

> The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the users' credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.

Reviewer 3: Why have you started adding `()` to the function names in the text here? There are many references to `signIn`, `signOut`, `showError` etc earlier on that don't have the signature part.

Author: This may be partially a revision issue and also one to do with a mix of code the developer writes versus code in the SDK. I'm not sure which direction to go on this and would love to hear what you think. In the above text I use the full signature of the SDK function because it's from the SDK and there are other "store" functions with different signatures.

> The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the users' credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.

Reviewer 3: ```suggestion
The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the user's credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.
```

Author: Nice catch "There can be only one." –Highlander. Change is in the next commit.

> The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the users' credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` which only tries to refresh the token if it's expired.

Author: I'll take "that" ;-). Change made and will be in the next commit.

> The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the user's credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.

Reviewer 2: ```suggestion
The `Credential` class manages the tokens for a user. The `signIn()` function that you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the user's credentials. A session exists if there's a default credential. Although there are calls to check if a credential is expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.
```

Add an `.onAppear` modifier to the main content view that checks for an existing session:

```swift
var body: some View {
   ZStack {
      ...
            ProgressView("Waiting")
         }
      }
      .onAppear() {
         busy = true
         // Check for an existing valid token.
         if let credential = Credential.default {
            Task {
               do {
                  // Make sure the token is valid.
                  try await credential.refreshIfNeeded()
                  // The token is valid, update the app state.
                  isSignedIn = true
               }
               catch {
                  // The user wasn't signed in or the token couldn't be refreshed.
                  // No action is required.
               }
               // The token is invalid, hide the activity indicator.
               busy = false
            }
         } else {

>             // There is existing session, hide the activity indicator.

Reviewer 2: ```suggestion
            // If there is an existing session, hide the activity indicator.
```

Author: A great catch, though it may be different text is required. The semantics of this else clause are that there is no existing session. Perhaps a comment of:
// There is no existing session, hide the activity indicator

Reviewer 2: makes perfect sense to me, thanks.
